pkg/scheduler/framework/v1alpha1/framework_test.go

```diff
@@ -585,6 +589,130 @@ func TestPreFilterPlugins(t *testing.T) {
 
 }
 
+func TestPlugins(t *testing.T) {
+	tests := []struct {
+		name     string
+		plugins  []*TestPlugin
+		wantCode Code
+	}{
+		{
+			name: "SuccessFilter",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{FilterStatus: int(Success)},
+				},
+			},
+			wantCode: Success,
+		},
+		{
+			name: "ErrorFilter",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{FilterStatus: int(Error)},
+				},
+			},
+			wantCode: Error,
+		},
+		{
+			name: "UnschedulableFilter",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{FilterStatus: int(Unschedulable)},
+				},
+			},
+			wantCode: Unschedulable,
+		},
+		{
+			name: "UnschedulableAndUnresolvableFilter",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj: injectedResult{
+						FilterStatus: int(UnschedulableAndUnresolvable)},
+				},
+			},
+			wantCode: UnschedulableAndUnresolvable,
+		},
+		// followings tests cover multiple-plugins scenarios
+		{
+			name: "ErrorAndSuccessFilters",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin1",
+					inj:  injectedResult{FilterStatus: int(Error)},
+				},
+
+				{
+					name: "TestPlugin2",
+					inj:  injectedResult{FilterStatus: int(Error)},
+				},
+			},
+			wantCode: Error,
+		},
+		{
+			name: "SuccessAndErrorFilters",
+			plugins: []*TestPlugin{
+				{
+
+					name: "TestPlugin1",
+					inj:  injectedResult{FilterStatus: int(Success)},
+				},
+				{
+					name: "TestPlugin2",
+					inj:  injectedResult{FilterStatus: int(Error)},
+				},
+			},
+			wantCode: Error,
+		},
+		{
+			name: "SuccessAndUnschedulableFilters",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin1",
+					inj:  injectedResult{FilterStatus: int(Success)},
+				},
+
+				{
+					name: "TestPlugin2",
+					inj:  injectedResult{FilterStatus: int(Unschedulable)},
+				},
+			},
+			wantCode: Unschedulable,
+		},
+	}
+
+	for _, tt := range tests {
+		registry := Registry{}
+		cfgPls := &config.Plugins{Filter: &config.PluginSet{}}
+		for _, pl := range tt.plugins {
+			// register all plugins
+			if err := registry.Register(pl.name,
+				func(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
+					return pl, nil
+				}); err != nil {
+				t.Errorf("fail to register filter plugin (%s)", pl.name)
+			}
+			// append plugins to filter pluginset
+			cfgPls.Filter.Enabled = append(
+				cfgPls.Filter.Enabled,
+				config.Plugin{Name: pl.name})
+		}
+
+		f, err := NewFramework(registry, cfgPls, emptyArgs)
+		if err != nil {
+			t.Errorf("fail to create framework: %s", err)
+		}
+		status := f.RunFilterPlugins(context.TODO(), nil, pod, nil)
+		if status.Code() != tt.wantCode {
+			t.Errorf("test(%s) failed: expect %s got %s", tt.name,
```

Comment:
Use t.Run so that you don't need to print the test name yourself, e.g.,
https://github.com/kubernetes/kubernetes/pull/85188/files#diff-665ae525463c8555435b2bc773dde556R528